cdap-app-templates/cdap-etl/cdap-etl-lib/src/main/java/co/cask/cdap/template/etl/batch/source/FileBatchSource.java

```diff
@@ -0,0 +1,211 @@
+import co.cask.cdap.template.etl.common.FileBatchProperties;
+import com.google.gson.Gson;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+/**
+ * A {@link BatchSource} to use any distributed file system as a Source.
+ */
+@Plugin(type = "source")
+@Name("FileBatchSource")
+@Description("Batch source for File Systems")
+public class FileBatchSource extends BatchSource<LongWritable, Object, StructuredRecord> {
+
+  public static final String INPUT_NAME_CONFIG = "input.path.name";
+  public static final String INPUT_REGEX_CONFIG = "input.path.regex";
+  public static final String LAST_TIME_READ = "last.time.read";
+  public static final String CUTOFF_READ_TIME = "cutoff.read.time";
+  public static final String USE_TIMEFILTER = "timefilter";
+
+  private static final String REGEX_DESCRIPTION = "Regex to filter out filenames in the path. " +
+    "To use the TimeFilter, input \"timefilter\". The TimeFilter assumes that it " +
+    "is reading in files with the File log naming convention of YYYY-MM-DD-HH-mm-SS-Tag. The TimeFilter " +
+    "reads in files from the previous hour if the timeTable field is left blank. So if it's currently " +
+    "2015-06-16-15 (June 16th 2015, 3pm), it will read in files that contain 2015-06-16-14 in the filename. " +
+    "If the field timeTable is present, then it will read files in that haven't been read yet.";
+  private static final String FILESYSTEM_PROPERTIES_DESCRIPTION = "JSON of the properties needed for the " +
+    "distributed file system. The formatting needs to be as follows:\n{\n\t\"propertyMap\" : {\n\t\t\"prop1\" : " +
+    "{\n\t\t\t\"propertyName\" : \"<insert property name>\" \n\t\t\t \"propertyValue\" : \"<insert property value>\"" +
+    " \n\t\t}, ...\n\t}\n}";
+  private static final String PATH_DESCRIPTION = "Path to file(s) to be read. If a directory is specified, " +
+    "terminate the path name with a \'/\'";
+  private static final String TABLE_DESCRIPTION = "Name of the Table that keeps track of the last time files " +
+    "were read in.";
+  private static final String INPUT_FORMAT_CLASS_DESCRIPTION = "Name of the input format class, which must be a " +
+    "subclass of FileInputFormat. Defaults to TextInputFormat";
+  private static final String FILESYSTEM_DESCRIPTION = "Distributed file system to read in from.";
+  private static final Gson GSON = new Gson();
+  private static final Logger LOG = LoggerFactory.getLogger(FileBatchSource.class);
+  private static final SimpleDateFormat DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd-HH-mm");
+  private static final Schema DEFAULT_SCHEMA = Schema.recordOf(
```

Comment:
Let's move this to ``ETLUtils.java`` for now, and re-use it in this class and in the ``StreamBatchSource`` (or wherever else it is used)